src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -942,6 +952,16 @@ public BoundExpression Typeof(TypeSymbol type)
             { WasCompilerGenerated = true };
         }
 
+        public BoundExpression TypeofBeforeRewriting(TypeSymbol type)
+        {
+            return new BoundTypeOfOperator(
+                Syntax,
+                Type(type),
+                // For some reason the rewriting logic asserts that the GetTypeFromHandle property is null.
+                null,
```

Comment:
It's probably cleaner to remove the assertion in the rewriting logic (and eliminate this method), but I didn't want to do that without the approval of someone in the compiler team.